sldns/str2wire.c

```diff
@@ -1495,7 +1500,37 @@ sldns_str2wire_svcbparam_alpn_value(const char* val,
 	sldns_write_uint16(rd + 2, dst_len);
 	memcpy(rd + 4, unescaped_dst, dst_len);
 	*rd_len = 4 + dst_len;
-	
+
+	return LDNS_WIREPARSE_ERR_OK;
+}
+
+static int
+sldns_str2wire_svcbparam_dohpath_value(const char* val,
+	uint8_t* rd, size_t* rd_len)
+{
+	uint8_t     unescaped_dst[65536];
+	uint8_t    *dst = unescaped_dst;
+	size_t      dst_len;
+	size_t 	    val_len;
+
+	val_len = strlen(val);
+
+	if (val_len > sizeof(unescaped_dst)) {
+		return LDNS_WIREPARSE_ERR_SYNTAX_INTEGER_OVERFLOW;
+	}
+
+	if (val_len > 255) {
+		return LDNS_WIREPARSE_ERR_SVCB_DOHPATH_KEY_TOO_LARGE;
+	}
+
+	dst_len = sldns_str2wire_svcbparam_parse_copy_unescaped(dst, val, val_len);
+
```

Comment:
This write could result in a buffer overflow, as the `rd_len` isn't checked before writing the data. It's missing:
```suggestion
	if (*rd_len < 4 + dst_len)
		return LDNS_WIREPARSE_ERR_BUFFER_TOO_SMALL;
```